include/ensmallen_bits/bigbatch_sgd/bigbatch_sgd_impl.hpp

```diff
@@ -244,13 +244,10 @@ BigBatchSGD<UpdatePolicyType>::Optimize(
       terminate |= Callback::BeginEpoch(*this, f, iterate, epoch,
           overallObjective, callbacks...);
 
-      // Reset the counter variables if we will continue.
-      if (i != actualMaxIterations)
-      {
-        lastObjective = overallObjective;
-        overallObjective = 0;
-        currentFunction = 0;
-      }
+      // Reset the counter variables.
+      lastObjective = overallObjective;
+      overallObjective = 0;
+      currentFunction = 0;
```

Comment:
Here is one other place where we should revert to the old implementation (see https://github.com/mlpack/ensmallen/pull/425#discussion_r2089631394).

```suggestion
      // Reset the counter variables if we will continue.
      if (i != actualMaxIterations)
      {
        lastObjective = overallObjective;
        overallObjective = 0;
        currentFunction = 0;
      }
```